src/Core/Services/ResolverMiddleware.cs

```diff
@@ -218,9 +219,20 @@ private static object PreParseLeaf(IMiddlewareContext context, string leafJson)
             };
         }
 
-        public static bool RepresentsNullValue(JsonElement element)
+        public static bool RepresentsNullValue(JsonElement element, IType fieldType)
         {
-            return string.IsNullOrEmpty(element.ToString()) && element.GetRawText() == "null";
+            bool dwNullCheck = false;
+            if (fieldType is not StringType)
+            {
+                // if the field asked for is not a string type but element return kind is a string
+                // check rawtext, if rawtext is empty then it is a null value.
+                if (element.ValueKind == JsonValueKind.String && string.IsNullOrEmpty(element.ToString()))
+                {
+                    dwNullCheck = true;
+                }
+            }
+
+            return (string.IsNullOrEmpty(element.ToString()) && element.GetRawText() == "null") || dwNullCheck;
```

Comment:
Can this be simplified to the following where new param is not needed?:
```suggestion
return element.ValueKind is not (JsonValueKind.Undefined or JsonValueKind.Null)
```

Reply:
For DW, since we return a string the jsonValueKind will be string. We need to check if the return type being asked is not a string. if return type is int and we are returning empty from db then we can infer that the ISNULL check replaced the null with empty string while constructing the json and hence returned null. For strings however, we wont be able to differentiate between empty and null. Could not think of a way around this for now.

Reply:
Can you give an example of the JSON contents of the JsonElement in this case?

Is this what you mean?
```json
{
"propertyName": "null"
}
```

Is this not possible?
```json
{
"propertyName": null
}
```

Reply:
it will return "propertyName": "", I chose to keep it string.empty but can be changed to null. Will not impact the string case. for all other datatypes we would have to make a check and determine if the db value is null.

Reply:
This is now fixed. using alternatives discussed in dab sync up meeting.